### PR TITLE
Update `source-foundations` version in `source-react-components` and other packages

### PR DESCRIPTION
**Pull request by joecowton1**
## What are you changing?

- Updating `@guardian/source-foundations` across all packages, and in turn updating all corresponding `@guardian` packages so we're using the latest versions with no peer dependency conflicts.
- This includes a number of small spacing and alignment fixes resulting from the line height changes in `@guardian/source-foundations`:
  - Updated `Accordion` component spacing: #757
  - Fixed label and supporting text alignment in `Checkbox` and `Radio` components: #758
  - Updated `Select` component to add missing spacing and bring in line with `TextInput`: #759 

## Why?

- We recently [updated line height in `@guardian/source-foundations`](https://github.com/guardian/csnx/pull/750). This should be reflected everywhere and any resulting regressions fixed.

### CODE REVIEW

File: libs/@guardian/atoms-rendering/package.json [hunk 1]
```diff
@@ -9,13 +9,13 @@
...
 		"@guardian/ab-core": "5.0.0",
-		"@guardian/commercial": "10.0.0",
-		"@guardian/consent-management-platform": "13.5.0",
-		"@guardian/eslint-plugin-source-foundations": "13.0.0",
-		"@guardian/eslint-plugin-source-react-components": "17.0.0",
-		"@guardian/libs": "15.1.0",
-		"@guardian/source-foundations": "12.0.0",
-		"@guardian/source-react-components": "15.0.1",
+		"@guardian/commercial": "10.8.0",
+		"@guardian/consent-management-platform": "13.6.1",
```
Comment (JamieB-gu) [inline libs/@guardian/atoms-rendering/package.json:13]: Will defer to @guardian/commercial-dev @guardian/transparency-consent on these two.

File: pnpm-lock.yaml [hunk 2]
```diff
@@ -192,13 +192,13 @@ importers:
...
+      '@guardian/commercial': 10.8.0_typescript@5.1.3
+      '@guardian/consent-management-platform': 13.6.1_uo3bjqllavv5kwoq2zqgzcz5h4
       '@guardian/eslint-plugin-source-foundations': link:../eslint-plugin-source-foundations
       '@guardian/eslint-plugin-source-react-components': link:../eslint-plugin-source-react-components
-      '@guardian/libs': 15.1.0_xfreuenalcno2zxheqz32kfzfe
-      '@guardian/source-foundations': 12.0.0_xfreuenalcno2zxheqz32kfzfe
-      '@guardian/source-react-components': 15.0.1_klugjjbb25o62o6wn2a7ow2ku4
+      '@guardian/libs': link:../libs
+      '@guardian/source-foundations': link:../source-foundations
+      '@guardian/source-react-components': link:../source-react-components
```
Comment (JamieB-gu) [inline pnpm-lock.yaml:201]: Just out of curiosity, what does this change do?

**Review by JamieB-gu — APPROVED** [review]
LGTM from the dotcom side.

File: pnpm-lock.yaml (hunk 2 above)
Comment (mxdvl) [reply pnpm-lock.yaml:201]: This looks like it might be incorrect, as packages should point to an immutable store, not a locally linked package?

Comment (sndrs) [reply pnpm-lock.yaml:201]: my hunch is that's pnpm knowing what it has locally and preferring it over remote?

**Comment by jamesmockett** [comment]
Akemi is reviewing these updates as reducing the line height has caused a number of small changes to spacing and alignment. Most of these are fine and completely expected, but it's likely we'll need to make a few styling tweaks before this is ready to be released.

File: libs/@guardian/atoms-rendering/package.json (hunk 1 above)
Comment (akinsola-guardian) [reply libs/@guardian/atoms-rendering/package.json:13]: Thanks for updating "@guardian/consent-management-platform".

**Review by akinsola-guardian — APPROVED** [review]

**Pull request merged**